borsh/src/de/mod.rs

```diff
@@ -400,6 +403,28 @@ where
     }
 }
 
+#[cfg(any(test, feature = "bytes"))]
+impl BorshDeserialize for bytes::Bytes {
+    #[inline]
+    fn deserialize_reader<R: Read>(reader: &mut R) -> Result<Self> {
+        let vec = <Vec<u8>>::deserialize_reader(reader)?;
+        Ok(vec.into())
+    }
+}
+
+#[cfg(any(test, feature = "bytes"))]
+impl BorshDeserialize for bytes::BytesMut {
+    #[inline]
+    fn deserialize_reader<R: Read>(reader: &mut R) -> Result<Self> {
+        let len = u32::deserialize_reader(reader)?;
+        let mut out = BytesMut::with_capacity(hint::cautious::<u8>(len));
+        for _ in 0..len {
+            out.put_u8(u8::deserialize_reader(reader)?);
+        }
```

Comment:
Should this follow the path implemented #129? i.e. do not allocate the whole chunk at once

Reply:
`hint::cautious` limits the initial allocation to 4kb, and then `BytesMut` uses its internal resizing strategy after that (presumable table doubling) - so I think we're protected from the issue in #129. Am I missing something?